src/Parser/Monk/Windwalker/Modules/Items/TheEmperorsCapacitor.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellLink from 'common/SpellLink';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Module from 'Parser/Core/Module';
+
+class TheEmperorsCapacitor extends Module {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    combatants: Combatants,
+  };
+  totalStacks = 0;
+  currentStacks = 0;
+  stacksUsed = 0;
+  stacksWasted = 0;
+  cjlCasts = 0;
+  averageStacksUsed = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.THE_EMPERORS_CAPACITOR.id);
+  }
+
+  on_byPlayer_applybuffstack(event) {
+    const spellId = event.ability.guid;
+    // You can only ever apply 1 stack at a time
+    if (SPELLS.THE_EMPERORS_CAPACITOR_STACK.id === spellId) {
+      this.totalStacks += 1;
+      if (this.currentStacks === 20) {
+        this.stacksWasted += 1;
+      }
+      else {
+        this.currentStacks += 1;
+      }
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.CRACKLING_JADE_LIGHTNING.id === spellId) {
+      this.stacksUsed += this.currentStacks;
+      this.currentStacks = 0;
+      this.cjlCasts += 1;
+      this.averageStacksUsed = this.stacksUsed / this.cjlCasts;
+    }
+  }
+
+  item() {
+    const cracklingJadeLightning = this.abilityTracker.getAbility(SPELLS.CRACKLING_JADE_LIGHTNING.id);
+    const damage = cracklingJadeLightning.damageEffective;
```

Comment:
For whatever reason, `damageEffective` from the AbilityTracker is actually just amount, when you want amount + absorbed. You can do `const damage = cracklingJadeLightning.damageEffective + cracklingJadeLightning.damageAbsorbed;` Really, AbilityTracker should be fixed because this is misleading.

Also, `damageEffective` and `damageAbsorbed` will be undefined if you never cast CJL, so you might need to check for that.

Reply:
I can move the damagecalculation to a class variable and then calculate it in a `on_byPlayer_damage` check